client/src/components/animations/AnimAction.js

```diff
@@ -0,0 +1,68 @@
+import { PropTypes } from "prop-types";
+import React, { useState, useEffect, useCallback, useRef } from "react";
+
+import AnimationLoop from "./AnimLoop";
+
+const randomTime = ([begin, end]) => 1000 * (begin + (end - begin) * Math.random());
+
+/**
+ * AnimationWithAction renders an infinite animation from a looped sprite,
+ * and sometimes (random generated within timeBetweenAction[0] and timeBetweenAction[1])
+ * if shows the action sprite just once
+ */
+const AnimationWithAction = ({ size, loopImage, actionImage, timeBetweenAction }) => {
+  const [isAction, setIsAction] = useState(false);
+  const timerIdRef = useRef(0);
```

Comment:
Why use a ref and not a state here? Since the reference doesn't link a dom element

Reply:
An update to the state triggers a re-render of the component. Here, we just want to have the value of the timer to clear it on unmounting : I see useRef as a pointer to this value.
I'm not 100% sure of this usage though so feel free to tell if it's not good!

Reply:
I believe a Ref is used to keep a DOM element in a variable, but its use is often controversial because it's not really React thinking. 
Here is not really the purpose, 
have you tried to just use a local variable ?

Reply:
Time to grrr

Reply:
Not this time aha I was just curious but if it works well we can keep it! :)

Reply:
According to the [docs](https://reactjs.org/docs/hooks-reference.html#useref):
> However, `useRef()` is useful for more than the ref attribute. It’s handy for keeping any mutable value around similar to how you’d use instance fields in classes.

The problem with a local variable (and I have tried it) it that in `useEffect()`, the value of the variable is read when the hook is ran, otherwise  the value is memoized. If we had only one timer, it would be fine, but here we are (re)setting the timer at multiple places.

Reply:
Oh yes, indeed it's perfect then!

Reply:
But grrrr to @valentinperignon because it's not the time to grrrr